### PR TITLE
Login

### PR DESCRIPTION
**Pull request by HerveJules**
## Task completed to this PR
- signup with encrypted password using bcrypt
- login and compare password to hashed password
- generating jwt which holds content for future use when needed

## Task pending on this PR
- passport authentication using jwt

## Incomplete task
- validations concerning login

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,11 @@
...
+
+#babel configuration file
+
+=======
 node_modules
 .env
-package-lock.json
\ No newline at end of file
+package-lock.json
+>>>>>>> 77e669c5f9638b50939b8cd25a55cdda5cddcc22
```
Comment (shemaeric) [inline .gitignore]: solve these errors in git ignore

File: src/controllers/user.js [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
               return res.status(400).send({
                 status:400,
                 errorMessage: 'The User with that email exists'
               })
             }
 
-            const userSave = await users.create({ email, password, jobtitle, tin});
+            const encryptedPassword = await enc(password);
+            const userSave = await users.create({ email, password:encryptedPassword, jobtitle, tin});
+            console.log(userSave.password);
```
Comment (shemaeric) [inline src/controllers/user.js]: remove this console log.

File: src/controllers/user.js [hunk 3]
```diff
@@ -28,12 +34,46 @@ class Users {
...
+      const Password = enc.compareHashedPassword(password,userfindOne.password);
+      if (Password === password) {
+        const user = {
+          id:userfindOne.id,
+          tin:userfindOne.tin
+        }
+        const token = jwt.sign(user,'secret');
+        return res.status(201).send({
+          status:201,
+          message:'logged in successfully',
```
Comment (shemaeric) [inline src/controllers/user.js]: The message should be descriptive like : You have successfully logged in.

File: src/controllers/user.js [hunk 4]
```diff
@@ -28,12 +34,46 @@ class Users {
...
+          token
+        })
+      } else {
+        return res.status(400).send({
+          status:401,
+          message:'incorrect password'
```
Comment (shemaeric) [inline src/controllers/user.js:64]: same, change the description of the message.

File: src/controllers/user.js [hunk 5]
```diff
@@ -28,12 +34,46 @@ class Users {
...
+        })
+      }
+      
+    }else{
+      // if no user found by email report incorrect email
+      return res.status(400).send({
+        status:400,
+        message:'incorrect email '
```
Comment (shemaeric) [inline src/controllers/user.js]: change the message description.

File: src/controllers/user.js [hunk 6]
```diff
@@ -28,12 +34,46 @@ class Users {
...
+        }
+        const token = jwt.sign(user,'secret');
+        return res.status(201).send({
+          status:201,
+          message:'logged in successfully',
+          token
+        })
+      } else {
+        return res.status(400).send({
+          status:401,
```
Comment (shemaeric) [inline src/controllers/user.js:63]: the status returned in body is not the same as returned in the response.

File: src/helpers/index.js [hunk 7]
```diff
@@ -0,0 +1,14 @@
...
+	return hashPassword;
+}
+
+const compareHashedPassword = (hash,password) => {
+		
+	return bcrypt.compareSync(password,hash);
+}
+	
+
+export default {generateHash,compareHashedPassword};
```
Comment (shemaeric) [inline src/helpers/index.js]: put the new line.

File: src/routes/user.js [hunk 8]
```diff
@@ -4,5 +4,5 @@ import User from '../controllers/user';
 const router = express.Router();
 
 router.post('/api/v1/auth/signup', User.createUser);
-
+router.post('/api/v1/auth/signin',User.auth);
 export default router;
```
Comment (shemaeric) [inline src/routes/user.js:10]: new line here too.

File: .babelrc [hunk 9]
```diff
@@ -1,3 +1,4 @@
+
 {
     "presets": ["@babel/preset-env"]
 }
```
Comment (shemaeric) [inline .babelrc:4]: new line again.

**Review by shemaeric — CHANGES_REQUESTED** [review]
You can apply the feedbacks and test again the functionality of the PR so that we can merge.

File: src/helpers/index.js (hunk 7 above)
Comment (HerveJules) [reply src/helpers/index.js]: what do you mean?
like this
```suggestion
export default {
 generateHash,
 compareHashedPassword
};
```

**Review by shemaeric — APPROVED** [review]

**Pull request merged**